app/src/app/classes/stix/relationship.ts

```diff
@@ -74,6 +81,10 @@ export class Relationship extends StixObject {
                 let serialized = this.serialize();
                 serialized.source_object = x.find(result => result.stix.id == new_source_ref);
                 this.deserialize(serialized);
+                var src_sub = this.get_parent_object(serialized.source_object, restAPIService).subscribe({
+                    next: (res) => { this.source_parent = res; },
+                    complete: () => { if (src_sub) src_sub.unsubscribe(); }
+                });
```

Comment:
Also, I realize your approach is to call this function and rely on the `EMPTY` response to weed out cases where there is no parent (in all cases except sub-techniques and data components), but it may be advantageous to simply not call the function at all in the case that it is not that sort of relationship. Unless I'm missing something that seems like simpler approach.

With the above refactor, you would probably need to construct the contents of the pipe dependent on the relationship type or source/target objects. Not quite sure how you'd do that, maybe using [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax)?